shell/platform/fuchsia/flutter/tests/fake_flatland_unittests.cc

```diff
@@ -0,0 +1,129 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "fakes/scenic/fake_flatland.h"
+
+#include <lib/async-testing/test_loop.h>
+#include <lib/async/dispatcher.h>
+
+#include <string>
+
+#include "flutter/fml/logging.h"
+#include "fuchsia/ui/composition/cpp/fidl.h"
+#include "gtest/gtest.h"
+
+namespace flutter_runner::testing {
+namespace {
+
+std::string GetCurrentTestName() {
+  return ::testing::UnitTest::GetInstance()->current_test_info()->name();
+}
+
+}  // namespace
+
+class FakeFlatlandTest : public ::testing::Test {
```

Comment:
Normally I'd agree with that.  There are two key things to keep in mind here:
+FakeFlatland is eventually going into the SDK, and it will be maintained by the scenic team apart from flutter or any other specific client.  I'm developing it here for now, but in particular scenic wants to use this for chromium<->scenic unit tests (they don't exist yet and that code is untested).
+The Scenic Fakes have some pretty subtle behavior around resource lifetime and reference counting that I haven't wired up for FakeFlatland yet: that's the crux of the TODO.
+This behavior, especially the ref counting, is specified as part of the Flatland API contract.  See here (written for gfx, but the core principle still applies): https://fuchsia.dev/fuchsia-src/concepts/graphics/scenic/scenic_resource_lifecycle?hl=en

So, my thinking here was: "write a test which verifies FakeFlatland *against the Flatland API contract* and also has 0 non-SDK dependencies"

Reply:
Makes sense, thanks for the detailed explanation!